test/unit/math/unaryOps/src/unaryOps.cpp

```diff
@@ -0,0 +1,398 @@
+                { &std::floor,  &alpaka::math::floor<TAcc,Data>,    Range::UNRESTRICTED      },
+                { &std::log,    &alpaka::math::log<TAcc,Data>,      Range::POSITIVE_ONLY     },
+                { &std::round,  &alpaka::math::round<TAcc,Data>,    Range::UNRESTRICTED      },
+                { &rsqrt<Data>, &alpaka::math::rsqrt<TAcc,Data>,    Range::POSITIVE_ONLY     },
+                { &std::sin,    &alpaka::math::sin<TAcc,Data>,      Range::UNRESTRICTED      },
+                { &std::sqrt,   &alpaka::math::sqrt<TAcc,Data>,     Range::POSITIVE_AND_ZERO },
+                { &std::tan,    &alpaka::math::tan<TAcc,Data>,      Range::UNRESTRICTED      },
+                { &std::trunc,  &alpaka::math::trunc<TAcc,Data>,    Range::UNRESTRICTED      }
+
+            };
+
+        Idx const numOps = sizeof(arr)/sizeof(TestStruct<TAcc,Data>);
+        Idx const elementsPerThread(1u);
+        Idx const sizeArgs(10u);
+        Idx const sizeRes= sizeArgs * numOps;
+
+        // Create the kernel function object.
+        UnaryOpsKernel kernel;
+
+        // Get the host device.
+        auto const devHost(
+            alpaka::pltf::getDevByIdx<PltfHost>(0u));
+
+        // Select a device to execute on.
+        auto const devAcc(
+            alpaka::pltf::getDevByIdx<PltfAcc>(0u));
+
+        // Get a queue on this device.
+        QueueAcc queue(devAcc);
+
+        alpaka::vec::Vec<Dim, Idx> const extent(numOps);
+
+        // Let alpaka calculate good block and grid sizes given our full problem extent.
+        alpaka::workdiv::WorkDivMembers<Dim, Idx> const workDiv(
+            alpaka::workdiv::getValidWorkDiv<TAcc>(
+                devAcc,
+                extent,
+                elementsPerThread,
+                false,
+                alpaka::workdiv::GridBlockExtentSubDivRestrictions::Unrestricted));
+
+
+        // Allocate host memory buffers.
+        auto memBufHostArgs(alpaka::mem::buf::alloc<Data, Idx>(devHost, sizeArgs));
+        auto memBufHostRes(alpaka::mem::buf::alloc<Data, Idx>(devHost, sizeRes));
+        auto memBufHostStructs(alpaka::mem::buf::alloc<TestStruct<TAcc, Data>, Idx>(devHost, extent));
+
+        Data * const pBufHostArgs = alpaka::mem::view::getPtrNative(memBufHostArgs);
+        Data * const pBufHostRes = alpaka::mem::view::getPtrNative(memBufHostRes);
+        TestStruct<TAcc,Data> * const pBufHostStructs = alpaka::mem::view::getPtrNative(memBufHostStructs);
+
+        // This is just for a better understanding which results are unchanged.
+        for(Idx i(0);i < numOps;++i)
+        {
+            for(Idx j(0);j < sizeArgs;++j)
+            {
+                pBufHostRes[j+i*sizeArgs] = -1;
+            }
+        }
+
+        // C++11 random generator for uniformly distributed numbers in {-100,..,100}
+        std::random_device rd{};
+        std::default_random_engine eng{ rd() };
+        std::uniform_real_distribution<Data> dist(0, 100);
```

Comment:
can we move the data generation to a separate file and share it between both new math unit tests